File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/contactnotifier/db/SentInvitation.java
package org.elastos.trinity.runtime.contactnotifier.db;

import android.database.Cursor;

public class SentInvitation {
    public String did;
    public String carrierAddress;

    /**
     * Creates a SentInvitation object from a SENT_INVITATIONS row.
     */
    static SentInvitation fromDatabaseCursor(Cursor cursor) {
        SentInvitation invitation = new SentInvitation();
        invitation.did = cursor.getString(cursor.getColumnIndex(DatabaseHelper.DID));
        invitation.carrierAddress = cursor.getString(cursor.getColumnIndex(DatabaseHelper.CARRIER_ADDRESS));
        return invitation;
    }
}

File: Trinity.Runtime/platform_src/ios/elastOS/Classes/ContactNotifier/db/SentInvitation.swift
import SQLite

public class SentInvitation {
    public var did: String
    public var carrierAddress: String
    
    public static let didField = Expression<String>(CNDatabaseHelper.DID)
    public static let carrierUserIdField = Expression<String>(CNDatabaseHelper.CARRIER_USER_ID)
    public static let carrierAddressField = Expression<String>(CNDatabaseHelper.CARRIER_ADDRESS)
    
    init() {
        did = ""
        carrierAddress = ""
    }

    /**
     * Creates a SentInvitation object from a SENT_INVITATIONS row.
     */
    static func fromDatabaseCursor(row: Row) -> SentInvitation {
        let invitation = SentInvitation()
        invitation.did = row[didField]
        invitation.carrierAddress = row[carrierAddressField]
        return invitation
    }
}
